modules/util/portal-tools-rest-builder/src/main/resources/com/liferay/portal/tools/rest/builder/dependencies/client_resource.ftl

```diff
@@ -159,7 +160,25 @@ public interface ${schemaName}Resource {
 					_logger.log(Level.WARNING, "HTTP response message: " + httpResponse.getMessage());
 					_logger.log(Level.WARNING, "HTTP response status code: " + httpResponse.getStatusCode());
 
-					throw new Problem.ProblemException(Problem.toDTO(content));
+					Problem.ProblemException problemException = null;
+
+					if (Objects.equals(httpResponse.getContentType(), "application/json")) {
+						problemException = new Problem.ProblemException(
+							Problem.toDTO(content));
+					}
+					else {
+						_logger.log(
+							Level.WARNING,
+							"Unable to process Content-Type: " + httpResponse.getContentType());
+
+						String jsonResponse =
```

Comment:
Avoid generating JSON inline. Instead instantiate a new Problem and set the status.